api/grpcserver/post_client.go

```diff
@@ -0,0 +1,101 @@
+package grpcserver
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	pb "github.com/spacemeshos/api/release/go/spacemesh/v1"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+)
+
+// postClient represents a connection to a PoST service.
+//
+// It uses the grpc interface of the node to send commands to the post service.
+// Additionally if instructed it will start the post service and connect it to
+// the node.
+type postClient struct {
+	service *PostService
+
+	closed chan struct{}
+}
+
+func newPostClient(service *PostService) *postClient {
+	return &postClient{
+		service: service,
+		closed:  make(chan struct{}),
+	}
+}
+
+func (pc *postClient) Proof(ctx context.Context, challenge []byte) (*types.Post, *types.PostMetadata, error) {
+	ch := make(chan *pb.ServiceResponse, 1)
```

Comment:
I prefer to name variables after they use, rather than by their type. How about
```suggestion
	response := make(chan *pb.ServiceResponse, 1)
```